src/core/storage/storage.types.ts

```diff
@@ -0,0 +1,139 @@
+import { v4 as uuidv4 } from "uuid";
+import { instanceToPlain, instanceToInstance } from "class-transformer";
+
+export type TagValue = string | boolean | undefined | Array<string> | null;
+export type TagsBase = {
```

Comment:
Can we just call this `Tags` and remove all the default, custom, base etc and simplify it. e.g. Aries Askar will just take in a JSON of tags and we won't have any default tags. I don't think we need the complexity